lib/di.dart

```diff
@@ -746,6 +749,21 @@ Future setup(
      return IoniaGiftCardDetailPage(getIt.get<IoniaGiftCardDetailsViewModel>(param1: giftCard));
   });
 
+  getIt.registerFactoryParam<IoniaMoreOptionsPage, List, void>((List args, _){
+    final giftCard = args.first as IoniaGiftCard;
+  
+    return IoniaMoreOptionsPage(giftCard); 
+  });
+
+  getIt.registerFactoryParam<IoniaCustomRedeemViewModel, IoniaGiftCard, void>((IoniaGiftCard giftCard, _) => IoniaCustomRedeemViewModel(giftCard));
+
+  getIt.registerFactoryParam<IoniaCustomRedeemPage, List, void>((List args, _){
+    final giftCard = args.first as IoniaGiftCard;
+  
+    return IoniaCustomRedeemPage(getIt.get<IoniaCustomRedeemViewModel>(param1: giftCard) ); 
+  });
```

Comment:
Please if you have pass only 1 or 2 params to factory then pass it directly not by List of arguments!